libs/payment/api/src/lib/payrexx/payrexx-client.ts

```diff
@@ -0,0 +1,62 @@
+import crypto from 'crypto'
+import qs from 'qs'
+import fetch, {Response} from 'node-fetch'
+
+export type PayrexxSuccessResponse<Data = any> = {
+  status: 'success'
+  data: Data
+}
+
+export type PayrexxErrorResponse = {
+  status: 'error'
+  message: string
+}
+
+export type PayrexxResponse<Data = any> = PayrexxSuccessResponse<Data> | PayrexxErrorResponse
+
+export class PayrexxClient {
+  constructor(
+    private readonly baseUrl: string,
+    private readonly instance: string,
+    private readonly secret: string
+  ) {}
+
+  private buildSignedQueryString(queryParams = {}) {
+    return qs.stringify({
+      ...queryParams,
+      ApiSignature: crypto
+        .createHmac('sha256', this.secret)
+        .update(qs.stringify(queryParams))
+        .digest('base64')
+    })
+  }
+
+  private buildBaseUrl(path) {
+    return this.baseUrl + path + '?instance=' + this.instance
+  }
+
+  async get<Data = any>(path, queryParams = {}): Promise<PayrexxResponse<Data>> {
+    const method = 'GET',
+      queryStrSigned = this.buildSignedQueryString(queryParams),
+      baseUrl = this.buildBaseUrl(path + '&' + queryStrSigned)
+    console.log({message: 'Payrexx GET request issued', path, queryParams})
```

Comment:
maybe we could use `logger()`

Reply:
I don't fully support the logger idea here. It complicates things and keeps logs flat strings. Modern logging should support objects. arrays, etc.
Let's talk about it !